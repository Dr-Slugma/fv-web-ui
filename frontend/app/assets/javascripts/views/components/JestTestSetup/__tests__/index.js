import React from 'react'
import ReactDOM from 'react-dom'
import { JestTestSetup } from '..'
// import renderer from 'react-test-renderer'

// Resolves promise on next tick:
// const flushPromises = () => {
//   return new Promise(resolve => {
//     setTimeout(resolve, 0)
//   })
// }
// Mock a dependency:
// jest.mock('../../some/lib', ()=> {
//   return {
//     aProp: {
//       aMethod: jest.fn(()=> Promise.resolve())
//     }
//   }
// })
// test('JestTestSetup Mounts', async () => {

// beforeEach(() => {
//   initializeCityDatabase();
// });

// afterEach(() => {
//   clearCityDatabase();
// });

test('JestTestSetup Mounts', () => {
  // const div = document.createElement('div')
  // await flushPromises()
  // expect(fakeHistory.push).toHaveBeenCalledTimes(1)
  // expect(utilsMock.posts.create).toHaveBeenCalledWith({date: expect.any(String)})
  // ReactDOM.render(<JestTestSetup />, div)

  // Structure: Arrange
  const handleSubmit = jest.fn()
  const container = document.createElement('div')
  ReactDOM.render(<JestTestSetup onSubmit={handleSubmit} />, container)

  const form = container.querySelector('form')
  const { username, password } = form.elements
  username.value = 'name from jest test'
  password.value = 'pasword from jest test'

  // form.dispatchEvent(new Event('submit'))
  form.dispatchEvent(new window.Event('submit'))

  // expect(container.querySelector('h1').textContent).toMatch('This is the form')

  expect(handleSubmit).toHaveBeenCalledTimes(1)
  expect(handleSubmit).toHaveBeenCalledWith({
    username: username.value,
    password: password.value,
  })
  // Structure: Act
  // Structure: Assert
})

test('automatic snapshot', () => {
  const handleSubmit = jest.fn()
  const container = document.createElement('div')
  ReactDOM.render(<JestTestSetup onSubmit={handleSubmit} />, container)

  const form = container.querySelector('form')
  const { username, password } = form.elements
  username.value = 'name from jest test'
  password.value = 'pasword from jest test'

  expect(form).toMatchSnapshot()
})
