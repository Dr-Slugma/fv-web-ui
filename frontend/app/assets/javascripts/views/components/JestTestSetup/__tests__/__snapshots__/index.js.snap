// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`automatic snapshot 1`] = `
<form
  class="JestTestSetup"
  data-reactid=".1"
  id="JestTestSetup"
>
  <label
    data-reactid=".1.0"
    for="username"
  >
    User name
  </label>
  <input
    data-reactid=".1.1"
    id="username"
    name="username"
    type="text"
  />
  <label
    data-reactid=".1.2"
    for="password"
  >
    Password
  </label>
  <input
    data-reactid=".1.3"
    id="password"
    name="password"
    type="text"
  />
</form>
`;
